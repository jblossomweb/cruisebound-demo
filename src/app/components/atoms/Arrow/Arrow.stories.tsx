import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import Arrow, { type ArrowProps } from './Arrow'

export default {
  title: 'components/atoms/Arrow',
  component: Arrow,
  argTypes: {
    color: { control: 'color' },
    size: { control: 'number' },
  },
} as Meta<typeof Arrow>

const Template: StoryFn<typeof Arrow> = (args: ArrowProps) => (
  <Arrow {...args} />
)

export const Default = Template.bind({})
Default.args = {}

export const Color = Template.bind({})
Color.args = {
  color: '#ff0000',
}

export const Sized = Template.bind({})
Sized.args = {
  size: 50,
}

export const Both = Template.bind({})
Both.args = {
  color: '#ff0000',
  size: 50,
}
