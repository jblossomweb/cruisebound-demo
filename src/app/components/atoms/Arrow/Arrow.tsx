/* eslint-disable react/self-closing-comp */
import React from 'react'

export interface ArrowProps {
  color?: React.CSSProperties['color']
  size?: number
}

const Arrow: React.FC<ArrowProps> = ({ color = '#0066FF', size = 9 }) => (
  <svg
    width={size}
    height={size}
    viewBox="0 0 9 9"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 4.5L8.5 4.5"
      stroke={color}
      strokeLinecap="round"
      strokeLinejoin="round"
    ></path>
    <path
      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
      stroke={color}
      strokeLinecap="round"
      strokeLinejoin="round"
    ></path>
  </svg>
)

export default Arrow
