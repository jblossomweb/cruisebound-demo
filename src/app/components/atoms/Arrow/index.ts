export type { ArrowProps } from './Arrow'
export { default } from './Arrow'
