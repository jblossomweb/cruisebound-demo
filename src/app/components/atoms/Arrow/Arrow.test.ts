import storySnaps from 'test/utils/storySnaps'
import * as stories from './Arrow.stories'

describe('app/components/atoms/Arrow', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
