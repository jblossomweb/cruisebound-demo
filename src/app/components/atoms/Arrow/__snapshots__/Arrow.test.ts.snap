// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Arrow stories Both matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="50"
    viewBox="0 0 9 9"
    width="50"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 4.5L8.5 4.5"
      stroke="#ff0000"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
      stroke="#ff0000"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;

exports[`app/components/atoms/Arrow stories Color matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="9"
    viewBox="0 0 9 9"
    width="9"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 4.5L8.5 4.5"
      stroke="#ff0000"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
      stroke="#ff0000"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;

exports[`app/components/atoms/Arrow stories Default matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="9"
    viewBox="0 0 9 9"
    width="9"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 4.5L8.5 4.5"
      stroke="#0066FF"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
      stroke="#0066FF"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;

exports[`app/components/atoms/Arrow stories Sized matches markup snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="50"
    viewBox="0 0 9 9"
    width="50"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 4.5L8.5 4.5"
      stroke="#0066FF"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
      stroke="#0066FF"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;
